lib/Controller/SMimeCertificatesController.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2022 Richard Steinmetz <richard@steinmetz.cloud>
+ *
+ * @author Richard Steinmetz <richard@steinmetz.cloud>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Mail\Controller;
+
+use OCA\Mail\Db\SMimeCertificate;
+use OCA\Mail\Db\SMimeCertificateMapper;
+use OCA\Mail\Exception\ServiceException;
+use OCA\Mail\Service\Attachment\UploadedFile;
+use OCA\Mail\Service\SMimeService;
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Db\DoesNotExistException;
+use OCP\AppFramework\Http\JSONResponse;
+use OCP\IRequest;
+
+class SMimeCertificatesController extends Controller {
+	private ?string $userId;
+	private SMimeService $certificateService;
+	private SMimeCertificateMapper $certificateMapper;
+
+	public function __construct(string $appName,
+								IRequest $request,
+								?string $userId,
+								SMimeService $certificateService,
+								SMimeCertificateMapper $certificateMapper) {
+		parent::__construct($appName, $request);
+		$this->userId = $userId;
+		$this->certificateService = $certificateService;
+		$this->certificateMapper = $certificateMapper;
+	}
+
+	/**
+	 * @NoAdminRequired
+	 * @TrapError
+	 *
+	 * @throws ServiceException
+	 */
+	public function index(): JSONResponse {
+		$certificates = $this->certificateMapper->findAll($this->userId);
+		foreach ($certificates as $certificate) {
+			$this->certificateService->decryptCertificateEntity($certificate);
```

Comment:
I don't like that this in place decryption will mark the cert entity as modified and if the resulting entity object accidentally gets passed to a mapper we will write the unencrypted value into the db. let me think about this.

Reply:
we could leave the property to always hold the encrypted value and only decrypt the values where we need it. this is how we do it for the IMAP password. it's decrypted when we instantiate the horde client and nowhere else. the decrypted value is saved to a local variable rather than fed back into the entity.

since we only do server-side s-mime for now, would it even be necessary to decrypt the certs for the frontend? shouldn't we even remove the certificate from json responses so they never leave the server? imagine you log in from someone else's computer. you don't want your private certs to ever be on that machine.

Reply:
The certificates are parsed on the client side to show more information. We could move this to the backend and thus eliminate the need to send certs to the frontend.

Reply:
Sounds good. Let's do that.

Reply:
Done